[Bryan-Goh] iP

Bryan-Goh opened the pull request:
# Adam
>“Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes/)
Adam is useful if you have these problem in your life:
- lonely
- **confused**
- *angry * ~~at~~  yourself
All you need todo is:
1. Study
2. Work
3. Use Adam
Features:
- [x] Keeps track of your task
- [ ] Keeps track of its completion rate(coming soon)
- [ ] Add delight to the experience when all tasks are complete (coming soon)
You can also use it to help code on this ```main``` method

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Bryan-Goh committed:
changed the name and implemented an initial skeletal version

Bryan-Goh committed:
Improve the skeletal version of Duke so that it echos commands entered by the user, and exits when the user types the command bye

Bryan-Goh committed:
Add, List

Bryan-Goh committed:
mark as done

Bryan-Goh committed:
ToDo, Event, Deadline

Bryan-Goh committed:
Automated Text UI Testing

Bryan-Goh committed:
Handle Errors

Bryan-Goh committed:
Delete

Bryan-Goh committed:
Add read and write

Bryan-Goh committed:
Add javadoc

Bryan-Goh committed:
Merge branch 'branch-level-7'

Bryan-Goh committed:
Add LocalDate

Bryan-Goh committed:
Add javadoc

Bryan-Goh committed:
Merge branch 'branch-level-8'

Bryan-Goh committed:
Add OOP

Bryan-Goh committed:
Implement packages

Bryan-Goh committed:
Merge commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552'

Bryan-Goh committed:
Add Gradle

Bryan-Goh committed:
Add JUnit Tests

Bryan-Goh committed:
Delete wrong javadoc

Bryan-Goh committed:
Add javadoc

Bryan-Goh committed:
Delete javadoc

Bryan-Goh committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/adam/Adam.java
#	src/main/java/adam/exception/AdamException.java
#	src/main/java/adam/exception/DateException.java
#	src/main/java/adam/exception/DeadlineException.java
#	src/main/java/adam/exception/EventsException.java
#	src/main/java/adam/exception/NoDescriptionException.java
#	src/main/java/adam/exception/NumberException.java
#	src/main/java/adam/exception/OneWordException.java
#	src/main/java/adam/exception/OutOfBoundsException.java
#	src/main/java/adam/tasks/Deadlines.java
#	src/main/java/adam/tasks/Events.java
#	src/main/java/adam/tasks/Task.java
#	src/main/java/adam/tasks/ToDos.java

Bryan-Goh committed:
Add proper coding standard

Bryan-Goh committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/adam/Adam.java
#	src/main/java/adam/Parser.java
#	src/main/java/adam/Storage.java
#	src/main/java/adam/TaskList.java
#	src/main/java/adam/Ui.java
#	src/main/java/adam/command/AddCommand.java
#	src/main/java/adam/command/EditCommand.java
#	src/main/java/adam/command/SingleCommand.java
#	src/main/java/adam/exception/AdamException.java
#	src/main/java/adam/exception/DateException.java
#	src/main/java/adam/exception/DeadlineException.java
#	src/main/java/adam/exception/EventsException.java
#	src/main/java/adam/exception/NoDescriptionException.java
#	src/main/java/adam/exception/NumberException.java
#	src/main/java/adam/exception/OneWordException.java
#	src/main/java/adam/exception/OutOfBoundsException.java
#	src/main/java/adam/tasks/Deadlines.java
#	src/main/java/adam/tasks/Events.java
#	src/main/java/adam/tasks/Task.java
#	src/main/java/adam/tasks/ToDos.java

Bryan-Goh committed:
Add find function

Bryan-Goh committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/adam/Adam.java
#	src/main/java/adam/Parser.java
#	src/main/java/adam/command/AddCommand.java
#	src/main/java/adam/command/EditCommand.java
#	src/main/java/adam/tasks/Deadlines.java
#	src/main/java/adam/tasks/Events.java
#	src/main/java/adam/tasks/Task.java
#	src/main/java/adam/tasks/ToDos.java

weeweh commented on src/main/java/adam/tasks/Deadline.java:
```diff
@@ -0,0 +1,27 @@
+package adam.tasks;
+
+import java.io.Serializable;
```
Could consider leaving a spacing between imports from different packages?

weeweh commented on src/main/java/adam/tasks/Deadline.java:
```diff
@@ -0,0 +1,27 @@
+ * This class is used for creating deadline tasks that has a due date
+ */
+public class Deadline extends Task implements Serializable {
+    protected LocalDate by;
```
Could add a JavaDoc comment for class fields?

weeweh commented on src/main/java/adam/Ui.java:
```diff
@@ -0,0 +1,138 @@
+     *
+     * @param size Size of the list.
+     */
+    public void getAmount(int size) {
```
Could be better named to reflect what amount it is getting

weeweh commented on src/main/java/adam/TaskList.java:
```diff
@@ -0,0 +1,167 @@
+ * This class holds all the methods that is realted to the task list such as adding and deleting tasks.
+ */
+public class TaskList {
+    private Ui ui = new Ui();
```
Why is TaskList constructing the UI again if it was constructed once in Adam.java?

weeweh reviewed (commented):
Just some minor changes to the imports.
Otherwise, LGTM

Bryan-Goh committed:
Add GUI

Bryan-Goh committed:
Add GUI to Adam

Bryan-Goh committed:
Add Greetings

Bryan-Goh committed:
Add bye

Bryan-Goh committed:
Change Javadoc

Bryan-Goh committed:
Merge branch 'brach-level-10'

# Conflicts:
#	src/main/java/adam/Adam.java

SynapseProgramming commented on src/main/java/adam/command/EditCommand.java:
```diff
@@ -0,0 +1,64 @@
+            respond = "Wrong input";
+        }
+        tasks.save(storage);
+        return respond;
```
Good job for applying the SLAP principle 👍

SynapseProgramming commented on src/main/java/adam/exception/AdamException.java:
Good job for creating multiple exceptions which are quite descriptive.

SynapseProgramming commented on src/test/java/adam/command/EditCommandTest.java:
good job for adding multiple unit tests :)

Bryan-Goh committed:
Update gradle.build

Bryan-Goh committed:
Add assert to 3 classes

To document important assumptions that should hold at various points in
the code.

Let's add some assert keyword in multiple methods that needs checking.

Using assert can add some additional safety and impact on
performance is also low.

Bryan-Goh committed:
Edit Javadoc

Bryan-Goh committed:
Merge pull request #2 from Bryan-Goh/branch-A-Assertions

Add assert to 3 classes

Bryan-Goh committed:
Extract 2 methods from the execute method

AddCommand class only has one method that exceeds 30 lines of code.

Let's extract 2 methods that return a string that can then be called by the execute method.

Doing this can make the code more readable.

Bryan-Goh committed:
Merge pull request #3 from Bryan-Goh/branch-A-CodeQuality

Extract 2 methods from the execute method

Bryan-Goh committed:
Add Tags

Bryan-Goh committed:
Add NoTagException

Bryan-Goh committed:
Merge branch 'branch-C-Tagging'

Bryan-Goh committed:
Update the Exception thrown

Bryan-Goh committed:
Add User Guide

Bryan-Goh committed:
Implement SLAP

Bryan-Goh committed:
Add JavaDocs

Bryan-Goh committed:
Edit layout of list

Bryan-Goh committed:
Edit import statements

Bryan-Goh committed:
Add Ui.png

Bryan-Goh committed:
Add Fat jar

Bryan-Goh committed:
Edit UG

Bryan-Goh committed:
Edit error checks

Bryan-Goh committed:
Edit jar name

Bryan-Goh committed:
Edit gradle buils

Bryan-Goh committed:
Add check styles
